engine-standalone-storage/src/sync/types.rs

```diff
@@ -759,9 +817,10 @@ enum BorshableTransactionKind<'a> {
     PausePrecompiles(Cow<'a, parameters::PausePrecompilesCallArgs>),
     ResumePrecompiles(Cow<'a, parameters::PausePrecompilesCallArgs>),
     Unknown,
+    SetEthConnectorContractAccount(Cow<'a, parameters::SetEthConnectorContractAccountArgs>),
```

Comment:
I'm worried about the order of the variants here. In borsh encoding the first byte is used to determine which variant of the enum to try to deserialize. But adding this variant above other, already established variants such as `SetOwner` and `PauseContract`, it changes the mapping of that first byte to the different variants. This is a breaking change because old transactions would not be deserialized correctly anymore.

But I'm not exactly sure how to resolve this because I know that silo deployments already existed and would have had some transactions with certain variant identifiers, and those might be different from those in the mainnet engine. If it is true that none of the methods after `SetEthConnectorContractAccount` have been called before then there is no backwards compatibility issue, but if they have been then we might have a conflict between slios and mainnet Aurora.

In the latter case I think mainnet Aurora should take priority since I think its transaction history is more important than silos (I don't think there are any produciton silos yet where the history is critical, but we should double check with @JonathanLogan ).

Reply:
Good catch 👍🏻 Definitely the`SetEthConnectorContractAccount` must be below existing transactions on the mainnet. Regarding breaking changes on the silo so I discussed it with Sergey and he said that we can make breaking changes in the silo smart contract since we don't have any smart contract in production. So I think it shouldn't be a problem.